src/components/FontUploader.tsx

```diff
@@ -31,17 +41,39 @@ const FontUploader: React.FC<{ onFontSelected: (selectedFont: File | null) => vo
       setSelectedFont(selectedFile);
       onFontSelected(selectedFile);
       setFontPreview(URL.createObjectURL(selectedFile));
+
+      const fileName = files[0].name;
+      const name = fileName.split('.').slice(0, -1).join('.');
+      setText(name);
+      fontUrl = URL.createObjectURL(files[0]);
     }
   };
 
   const handleDragOver = (e: React.DragEvent<HTMLDivElement>) => {
     e.preventDefault();
   };
 
+  const fontStyles: React.CSSProperties = {
+    fontFamily: side === 'left' ? "'CustomFontLeft', sans-serif" : "'CustomFontRight', sans-serif",
+  };
+
+  const fontFace = `
+  @font-face {
+    font-family: '${side === 'left' ? 'CustomFontLeft' : 'CustomFontRight'}';
+    src: url(${fontUrl}) format('opentype');
+    font-display: swap;
+  }
+`;
+
   return (
     <div className="font-uploader">
       <div className="drop-area" onDrop={handleDrop} onDragOver={handleDragOver}>
-        <p>Drag and drop a font file here</p>
+        <div>
+          <style>{fontFace}</style>
```

Comment:
This might be due to my lack of understanding of TypeScript, and I also understand that this approach isn’t new to this PR. 

But why are we regenerating the subtree whenever a new font is loaded? Why not just find the elements and set their `font-family` property? 🤨

Reply:
Perhaps something to address in another branch. I can open an issue if appropriate.

Reply:
Yea there might be an improved way, maybe someone could add onto the code in the future